network/p2p/middleware.go

```diff
@@ -360,6 +361,32 @@ func (m *Middleware) Unsubscribe(channel network.Channel) error {
 	return nil
 }
 
+// processAuthenticatedMessage processes a message and a source (indicated by its PublicKey) and eventually passes it to the overlay
+func (m *Middleware) processAuthenticatedMessage(msg *message.Message, originKey crypto.PublicKey) {
```

Comment:
I found still a bit confusing. 

It sounds like the message has been authenticated, but actually we are going to check the authentication, right?

If so, would `checkOriginAndProcess` be better?

Reply:
> It sounds like the message has been authenticated,

It has, the message has a clear origin, which is the signer of its envelope, which signature has been checked.

> we are going to check the authentication

No, we are going to check the protocol-level authorship claim against the prior authentication.

Reply:
Thanks, I see. would be great to add these knowledges into comments